### PR TITLE
Refactor do post for generator

### PR DESCRIPTION
**Pull request by annyhe**
- move the doPost code to generator controller
- remove generator specific parts from postUtils

### CODE REVIEW

File: api/v1/helpers/verbs/postUtils.js [hunk 1]
```diff
@@ -65,14 +60,6 @@ function handlePostResult(o, resultObj, props, res, req) {
       realtimeEvents.sample.add, props.associatedModels.aspect);
```
Comment (annyhe) [inline api/v1/helpers/verbs/postUtils.js]: @shriramshankar do we need the publish code here? Will samples code ever reach api/v1/helpers/verbs/postUtils.js?

File: api/v1/controllers/generators.js [hunk 2]
```diff
@@ -75,7 +77,50 @@ module.exports = {
...
+
+      // order collectors by name
+      if (o.collectors) {
+        u.sortArrayObjectsByField(o.collectors, 'name');
+      }
+
+      res.status(constants.httpStatus.CREATED).json(
+          u.responsify(o, helper, req.method));
+    })
+    .catch((err) => u.handleError(next, err, helper.modelName));
```
Comment (shriramshankar) [inline api/v1/controllers/generators.js]: Let's talk about these changes tomorrow.

File: api/v1/helpers/verbs/postUtils.js (hunk 1 above)
Comment (shriramshankar) [reply api/v1/helpers/verbs/postUtils.js]: Nice catch. This is not required.

File: api/v1/controllers/generators.js [hunk 3]
```diff
@@ -75,7 +77,50 @@ module.exports = {
...
+          .catch(() => resolve(postBody));
+        } else {
+          resolve(postBody);
+        }
+      });
+    }
+
+    getPostBody()
+    .then((_toPost) => helper.model.createWithCollectors(_toPost,
+      u.whereClauseForNameInArr, u.sortArrayObjectsByField))
```
Comment (shriramshankar) [inline api/v1/controllers/generators.js]: There is no need to pass "u.sortArrayObjectsByField" here.

File: api/v1/controllers/generators.js [hunk 4]
```diff
@@ -75,7 +77,50 @@ module.exports = {
...
+    .then((o) => featureToggles.isFeatureEnabled('returnUser') ?
+        o.reload() : o)
+    .then((o) => {
+      resultObj.dbTime = new Date() - resultObj.reqStartTime;
+      u.logAPI(req, resultObj, o);
+
+      // order collectors by name
+      if (o.collectors) {
+        u.sortArrayObjectsByField(o.collectors, 'name');
```
Comment (shriramshankar) [inline api/v1/controllers/generators.js]: Don't need to check for o.cleectors here. We can handle the undefined case in  "u.sortArrayObjectsByField".

**Comment by annyhe** [comment]
@shriramshankar the PR is updated as per the comments.
@iamigo the heroku CI is failing due to redis and postgres not being setup. It doesn't look like the fail is holding back the approval/merge.

File: api/v1/helpers/verbs/utils.js [hunk 5]
```diff
@@ -102,11 +102,16 @@ function handleUpdatePromise(resultObj, req, retVal, props, res) {
 
 /**
  * Sorts an array in-place.
+ * If given invalid input, do nothing.
  *
  * @param {Array} arr Array of objects
  * @param {String} fieldName The field to sort by
  */
 function sortArrayObjectsByField(arr, fieldName) {
+  if (!arr || !Array.isArray(arr) || !arr.length) {
```
Comment (iamigo) [inline api/v1/helpers/verbs/utils.js]: I would rewrite this as "positive" if condition...

```
if (arr && Array.isArray(arr) && arr.length) {
  arr.sort((a, b) => a[fieldName].localeCompare(b[fieldName]));
}
```

File: api/v1/controllers/generators.js (hunk 2 above)
Comment (annyhe) [reply api/v1/controllers/generators.js]: implemented the discussed changes

**Pull request merged**